packages/backend-secret/src/secret.ts

```diff
@@ -84,3 +85,15 @@ export const getSecretClient = (
     })
   );
 };
+
+/**
+ * Creates an Amplify secret client with inbuilt amplify error handling
+ * Used in the amplify CLI commands such as sandbox and pipeline-deploy
+ */
+export const getSecretClientWithAmplifyErrorHandling = (
```

Comment:
withAmplifyErrorHandling is a code smell; if AmplifyErrorHandling is the default behavior that we want, this should just be getSecretClient. The other one is the exception to this rule, and that's why I think getSecretClient -> getSecretClientForBackendLambda/getSecretClientWithoutAmplifyErrorHandling ; and this one becomes just 'getSecretClient'

Reply:
> withAmplifyErrorHandling is a code smell

Agreed.

> that's why I think getSecretClient -> getSecretClientForBackendLambda/getSecretClientWithoutAmplifyErrorHandling ; and this one becomes just 'getSecretClient'

Changing type of exception thrown could. constitute runtime breaking change (think of people who wrote catch clauses). 
It would be safer to log work item to revise this when we ship MV of this package. And perhaps have only one variant with amplify errors.